torch_xla/csrc/tensor_methods.cpp

```diff
@@ -1886,6 +1887,15 @@ std::tuple<XLATensorPtr, XLATensorPtr, XLATensorPtr> native_batch_norm_backward(
                          std::move(grad_bias));
 }
 
+std::tuple<XLATensorPtr, XLATensorPtr> native_dropout(
+    const XLATensorPtr& input, double p, c10::optional<bool> train) {
+  torch::lazy::NodePtr node = torch::lazy::MakeNode<NativeDropout>(
+      input->GetIrValue(), p, train,
+      XLAGraphExecutor::Get()->GetRngSeed(input->GetDevice()));
+  return std::make_tuple(input->CreateFrom(torch::lazy::Value(node, 0)),
+                         input->CreateFrom(torch::lazy::Value(node, 1)));
```

Comment:
Since the result torch tensor will be built from Lazy IR node, we need to set the correct type here. we can pass an additional arg for dtype at https://github.com/pytorch/xla/blob/a0a3aba9e5263e5aafd6e7895f36e2703ed3c4cd/torch_xla/csrc/tensor.h#L167

We should pass `at::ScalarType::Bool` for mask